### ISSUE TITLE
Accept JSON as data input

### ISSUE THREAD

**Issue by JekRock** [mojoaxel/d3-sunburst#6]
It would be nice to have the ability to pass a JSON data source as well as CSV

**Comment by mojoaxel** [comment]
JSON would be nice but this component is not maintained anymore.
I recommend using a simple "JSON to CSV" function!

**Comment by JekRock** [comment]
Got it, thanks.
Maybe it's better to mark the repository as "archived" if it's not maintained? Or just put this info to the README

**Comment by mojoaxel** [comment]
> Maybe it's better to mark the repository as "archived" if it's not maintained?

I was thinking about it but I still answer to issues and pull-requests are still also very welcome!

**Comment by mojoaxel** [comment]
@JekRock What JSON format would you recommend for this kind of data?

```csv
account-home-other-account-account-account,58
account-home-other-account-account-end,11
account-home-other-account-account-home,16
account-home-other-account-account-other,12
account-home-other-account-account-product,26
account-home-other-account-account-search,5
```

**Comment by puripant** [comment]
I assume the format is either:
```
{
  account-home-other-account-account-account: 58,
  account-home-other-account-account-end: 11,
  account-home-other-account-account-home: 16,
  account-home-other-account-account-other: 12,
  account-home-other-account-account-product: 26,
  account-home-other-account-account-search: 5
}
```
or the result of `buildHierarchy()`. I can add `loadJson()` and an example for either version.

**Comment by JekRock** [comment]
@mojoaxel I can think of two options:
1. Make a JSON wrapper around the current data format. For example
```json
{
  "data": [
    {
      "path": "account-home-other-account-account-account",
      "value": 58
    },
          ]
}
```
It shouldn't be different from processing the current CSV data structure. Also, it's easy to read and correct "by hand".

2. Make a graph-like data structure. For example
```json
{
  "data": [
    {
      "name": "account",
      "value": 10,
      "children": [
        {
          "name": "home",
          "value": 3,
          "children": []
        },
        {
          "name": "other",
          "value": 4,
          "children": []
        }
      ]
     }
  ]
}
```

That would probably require changing the parsing process. It's not easy no change "by hand". But the parsing process and data generation should be easier as it's a serialized graph.

**Comment by JekRock** [comment]
@mojoaxel Also, check out the data format that is used in Plotly for the same type of chart https://plotly.com/javascript/sunburst-charts/

**Comment by mojoaxel** [comment]
Actually I don't think we need a new function for this at all.

But a new example could be useful!

```js
sunburst.createVisualization({
  name: "root",
  children: [{
    name: "child1",
    children: [{
      name: "child1.1",
      size: 12
    }]
  }, {
    name: "child2",
    children: [{
      name: "child2.1",
      size: 34
    }]
  }]
});

**Comment by mojoaxel** [comment]
> Also, check out the data format that is used in Plotly for the same type of chart https://plotly.com/javascript/sunburst-charts/

If anybody also would like to add support for the plotly-format - be my guest!

**Comment by mojoaxel** [comment]
this has been implemented by @puripant with #9 :tada: 

If anybody still wants to import the (strange) plotly format. Feel free to create a merge request!
